app/src/main/java/ch/ubique/notifyme/app/checkin/CheckInFragment.java

```diff
@@ -0,0 +1,111 @@
+package ch.ubique.notifyme.app.checkin;
+
+import android.os.Bundle;
+import android.view.View;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.appcompat.widget.Toolbar;
+import androidx.fragment.app.Fragment;
+import androidx.fragment.app.FragmentManager;
+import androidx.fragment.app.FragmentTransaction;
+import androidx.lifecycle.ViewModelProvider;
+
+import com.google.android.material.button.MaterialButton;
+import com.google.android.material.button.MaterialButtonToggleGroup;
+
+import org.crowdnotifier.android.sdk.model.VenueInfo;
+
+import ch.ubique.notifyme.app.MainFragment;
+import ch.ubique.notifyme.app.MainViewModel;
+import ch.ubique.notifyme.app.R;
+import ch.ubique.notifyme.app.model.ReminderOption;
+import ch.ubique.notifyme.app.utils.NotificationHelper;
+import ch.ubique.notifyme.app.utils.ReminderHelper;
+import ch.ubique.notifyme.app.utils.VenueTypeIconHelper;
+
+public class CheckInFragment extends Fragment {
+
+	public final static String TAG = CheckInFragment.class.getCanonicalName();
+
+	private MainViewModel viewModel;
+	private VenueInfo venueInfo;
+
+	public CheckInFragment() { super(R.layout.fragment_check_in); }
+
+	public static CheckInFragment newInstance() {
+		return new CheckInFragment();
+	}
+
+
+	@Override
+	public void onCreate(@Nullable Bundle savedInstanceState) {
+		super.onCreate(savedInstanceState);
+		viewModel = new ViewModelProvider(requireActivity()).get(MainViewModel.class);
+		if (viewModel.getCheckInState() != null) {
+			venueInfo = viewModel.getCheckInState().getVenueInfo();
+		} else {
+			getActivity().getSupportFragmentManager().popBackStack(MainFragment.TAG, FragmentManager.POP_BACK_STACK_INCLUSIVE);
```

Comment:
Why the `getActivity().getSupportFragmentManager()` everywhere? Wouldn't `getParentFragmentManager()` work?